aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/ReadHandler.java

```diff
@@ -22,28 +21,40 @@
             final ProxyClient<RdsClient> proxyClient,
             final RequestLogger requestLogger
     ) {
-        return proxy.initiate("rds::read-db-parameter-group", proxyClient, request.getDesiredResourceState(), callbackContext)
-                .translateToServiceRequest(Translator::describeDbParameterGroupsRequest)
-                .backoffDelay(CONSTANT)
-                .makeServiceCall((describeDbParameterGroupsRequest, proxyInvocation) -> proxyInvocation.injectCredentialsAndInvokeV2(describeDbParameterGroupsRequest, proxyInvocation.client()::describeDBParameterGroups))
-                .handleError((describeDbParameterGroupsRequest, exception, client, resourceModel, ctx) ->
-                        Commons.handleException(
-                                ProgressEvent.progress(resourceModel, ctx),
-                                exception,
-                                DEFAULT_DB_PARAMETER_GROUP_ERROR_RULE_SET
-                        ))
+        return describeDbParameterGroups(proxy, proxyClient, request.getDesiredResourceState(), callbackContext)
                 .done((describeDbParameterGroupsRequest, describeDbParameterGroupsResponse, proxyInvocation, model, context) -> {
                     try {
                         final DBParameterGroup dBParameterGroup = describeDbParameterGroupsResponse.dbParameterGroups().stream().findFirst().get();
-                        final Set<Tag> tags = Tagging.listTagsForResource(proxyInvocation, dBParameterGroup.dbParameterGroupArn());
-                        return ProgressEvent.success(Translator.translateFromDBParameterGroup(dBParameterGroup, tags), context);
+                        context.setDbParameterGroupArn(dBParameterGroup.dbParameterGroupArn());
+                        return ProgressEvent.progress(Translator.translateFromDBParameterGroup(dBParameterGroup), context);
                     } catch (Exception exception) {
                         return Commons.handleException(
                                 ProgressEvent.progress(model, context),
                                 exception,
-                                SOFT_FAIL_TAG_DB_PARAMETER_GROUP_ERROR_RULE_SET
+                                DEFAULT_DB_PARAMETER_GROUP_ERROR_RULE_SET
                         );
                     }
-                });
+                })
+                .then(progress -> readTags(proxyClient, progress));
     }
+
+    protected ProgressEvent<ResourceModel, CallbackContext> readTags(
+            final ProxyClient<RdsClient> proxyClient,
+            final ProgressEvent<ResourceModel, CallbackContext> progress) {
+        ResourceModel model = progress.getResourceModel();
+        CallbackContext context = progress.getCallbackContext();
+        try {
+            String arn = progress.getCallbackContext().getDbParameterGroupArn();
```

Comment:
I think we might need to check its presence here or at least assert it is not null

Reply:
Done.